ibm/resource_ibm_private_dns_record.go

```diff
@@ -0,0 +1,193 @@
+		},
+
+		CustomizeDiff: customdiff.Sequence(
+			func(diff *schema.ResourceDiff, v interface{}) error {
+				return resourceTagsCustomizeDiff(diff)
+			},
+		),
+
+		Schema: map[string]*schema.Schema{
+			pdnsInstanceID: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			pdnsZoneID: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			pdnsRecordTTL: {
+				Type:     schema.TypeInt,
+				Required: true,
+				DefaultFunc: func() (interface{}, error) {
+					return 900, nil
+				},
+			},
+
+			pdnsRecordName: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			pdnsIPv4Address: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			pdnsRecordType: {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: func(val interface{}, field string) (warnings []string, errors []error) {
+					value := val.(string)
+					for _, rtype := range allowedPrivateDomainRecordTypes {
+						if value == rtype {
+							return
+						}
+					}
+
+					errors = append(
+						errors,
+						fmt.Errorf("%s is not one of the valid domain record types: %s",
+							value, strings.Join(allowedPrivateDomainRecordTypes, ", "),
+						),
+					)
+					return
+				},
+			},
+		},
+	}
+}
+func resourceIBMPrivateDnsRecordCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+
+	instanceID := d.Get(pdnsInstanceID).(string)
+	zoneID := d.Get(pdnsZoneID).(string)
+	recordType := d.Get(pdnsRecordType).(string)
+	ttl := d.Get(pdnsRecordTTL).(int)
+	ipv4 := d.Get(pdnsIPv4Address).(string)
+	name := d.Get(pdnsRecordName).(string)
+
+	createResourceRecordOptions := sess.NewCreateResourceRecordOptions(instanceID, zoneID)
+	createResourceRecordOptions.SetName(name)
+	createResourceRecordOptions.SetType(recordType)
+	createResourceRecordOptions.SetTTL(int64(ttl))
+	resourceRecordAData, _ := sess.NewResourceRecordInputRdataRdataARecord(ipv4)
+	createResourceRecordOptions.SetRdata(resourceRecordAData)
+	response, _, err := sess.CreateResourceRecord(createResourceRecordOptions)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s/%s", instanceID, zoneID, *response.ID))
+
+	return resourceIBMPrivateDnsRecordRead(d, meta)
+}
+
+func resourceIBMPrivateDnsRecordRead(d *schema.ResourceData, meta interface{}) error {
+
+	sess, err := meta.(ClientSession).PrivateDnsClientSession()
+	if err != nil {
+		return err
+	}
+
+	id_set := strings.Split(d.Id(), "/")
+	getResourceRecordOptions := sess.NewGetResourceRecordOptions(id_set[0], id_set[1], id_set[2])
+	_, _, err = sess.GetResourceRecord(getResourceRecordOptions)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceIBMPrivateDnsRecordUpdate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
We have support for update also?
https://cloud.ibm.com/docs/dns-svcs?topic=dns-svcs-cli-plugin-dns-services-cli-commands#update-resource-record